pkg/cloudprovider/provider/baremetal/plugins/tinkerbell/metadata/client.go

```diff
@@ -54,7 +55,7 @@ type AuthConfig struct {
 }
 
 type Client interface {
-	GetMachineMetadata() (*MachineMetadata, error)
+	GetMachineMetadata(ctx context.Context) (*MachineMetadata, error)
```

Comment:
Please don't break public APIs like that in a "simple go upgrade" PR. At least let's have a dedicated PR (or better not break at all).

Reply:
Good point, basically it was to avoid a linter error. Let me re-evalute this.

Reply:
Updated PTAL, I'll change the API in a follow up PR.